web/views.py

```
@@ -4700,6 +4702,286 @@
+    except Session.DoesNotExist:
+        return JsonResponse({"success": False, "message": "Session not found"}, status=404)
+    except User.DoesNotExist:
+        return JsonResponse({"success": False, "message": "Student not found"}, status=404)
+    except Exception as e:
+        return JsonResponse({"success": False, "message": f"Error: {str(e)}"}, status=500)
+
+
+@login_required
+def get_student_attendance(request):
+    """Get a student's attendance data for a specific course."""
+    if not request.headers.get("X-Requested-With") == "XMLHttpRequest":
```

Ruff (0.8.2)

Use `request.headers.get("X-Requested-With") != "XMLHttpRequest"` instead of `not request.headers.get("X-Requested-With") == "XMLHttpRequest"`

Replace with `!=` operator

(SIM201)